lib/swagger_yard/type.rb

```diff
@@ -77,13 +77,31 @@ def to_h
       if array?
         { "type" => "array", "items" => Type.from_type_list([array.join("<")]).to_h }
       elsif object?
-        {
-          "type" => "object",
-          "additionalProperties" => Type.from_type_list([object.join("<")]).to_h
-        }
+        parse_object
       else
         type
       end
     end
+
+    def parse_object
+      properties = {}
+      additional = nil
+      type_hash  = { "type" => "object" }
+
+      object.join("<").split(/,\s?/).each do |item|
+        key, *rest = item.split(": ")
```

Comment:
The rational for the space was to avoid conflicts with Ruby name spacing for references, e.g. `MyApp::Greeting`. It could be possible to use a regex with a negative look-ahead to split on a colon that's not followed by another colon.